feat(twitch): Add integrations for Twitch patches

ThePBone opened the pull request:
Used by:
* https://github.com/revanced/revanced-patches/pull/1074
* https://github.com/revanced/revanced-patches/pull/1075

oSumAtrIX reviewed (changes requested):
This PR adds quite a lot of boilerplate duplicate code. If possible reduce it.

ThePBone commented:
I cleaned it up a bit and removed some unnecessary code. There is some overlap with the utility classes from YouTube/Tiktok integrations, but I didn't import their classes to prevent confusion.

oSumAtrIX commented on app/src/main/java/app/revanced/twitch/settingsmenu/SettingsHooks.java:
```diff
@@ -0,0 +1,110 @@
+            LogHelper.debug("Revanced settings not requested");
+            return false; // User wants to enter another settings fragment
+        }
+        LogHelper.debug("Revanced settings requested");
```
```suggestion
        LogHelper.debug("ReVanced settings requested");
```

oSumAtrIX commented on app/src/main/java/app/revanced/twitch/settingsmenu/SettingsHooks.java:
```diff
@@ -0,0 +1,110 @@
+    private static final String EXTRA_REVANCED_SETTINGS = "app.revanced.twitch.settings";
+
+    /**
+     * Launches SettingsActivity and show revanced settings
```
```suggestion
     * Launches SettingsActivity and show ReVanced settings
```

oSumAtrIX commented on app/src/main/java/app/revanced/twitch/settingsmenu/SettingsHooks.java:
```diff
@@ -0,0 +1,110 @@
+     * Launches SettingsActivity and show revanced settings
+     */
+    public static void startSettingsActivity() {
+        LogHelper.debug("Launching revanced settings");
```
```suggestion
        LogHelper.debug("Launching ReVanced settings");
```

oSumAtrIX commented on app/src/main/java/app/revanced/twitch/settingsmenu/SettingsHooks.java:
```diff
@@ -0,0 +1,110 @@
+     * Helper for easy access in smali
+     * @return Returns string resource id
+     */
+    public static int getRevancedSettingsString() {
```
Probably want to use the casing of `ReVanced` here.

ThePBone replied:
Fixed with 2665f5a
